astropy/io/fits/hdu/hdulist.py

```diff
@@ -563,7 +565,7 @@ def pop(self, index=-1):
 
         Returns
         -------
-        hdu : HDU object
+        hdu : `_BaseHDU` object
```

Comment:
I guess one issue is that `_BaseHDU` is not in the documentation so it cannot be referenced ?

Reply:
yes, there are number of issues like this where I wanted to avoid adding to the ignore or the glossary, but ended up having to reference a hidden object. It's technically fine, but not ideal.

Reply:
If Sphinx does not complain about it that's fine for now.  But this is probably a case where adding a term in the xref config or glossary can be useful ?

Reply:
I personally never use FITS stuff, so I don't know what in the package merits a full glossary term versus options like advocating that the bases-class be public-scoped.

Reply:
This is used everywhere where referring to a generic HDU (rather than specifically to a `PrimaryHDU`, `ExtensionHDU`, BinTableHDU` etc.)? Since it is _the_ fundamental data structure, I also think something like a `_BaseHDU` entry in the glossary would be warranted.

Reply:
With a consensus opinion, I've added a glossary term for Base HDU and pointed it toward the correct module (since `_BaseHDU` is not in `__all__` I can't point it to the class).

Reply:
✅